python/lsst/ts/dream/csc/dream_csc.py

```diff
@@ -231,65 +235,73 @@ async def weather_loop(self) -> None:
                     self.log.debug(f"Connected to ESS:{self.config.ess_index}.")
 
                     self.weather_ok_flag = None
-                    self.last_weather_update = 0.0
                     if self.ess_remote is None:
                         self.log.error("Failed to connect to weather CSC.")
                         continue
 
+                    # Wait for the CSC to establish its connection.
+                    await asyncio.sleep(10.0)
+
                 try:
                     # Get weather data.
                     weather_ok_flag = True
                     air_flow = await self.ess_remote.tel_airFlow.next(
-                        flush=False, timeout=2
+                        flush=True,
+                        timeout=10,
                     )
                     if air_flow is None or air_flow.speed > 25:
                         weather_ok_flag = False
 
-                    precipitation = await self.ess_remote.evt_precipitation.get()
+                    precipitation = await self.ess_remote.evt_precipitation.aget(
+                        timeout=10
```

Comment:
Magic number, please extract to a constant. Either reuse the one defined above or a new one if the value is different.